pkg/ddl/ingest/engine_mgr.go

```diff
@@ -101,19 +101,31 @@ func (bc *litBackendCtx) Register(indexIDs []int64, uniques []bool, tblInfo *mod
 	return ret, nil
 }
 
-// UnregisterEngines implements BackendCtx.
-func (bc *litBackendCtx) UnregisterEngines() {
+// FinishAndUnregisterEngines implements BackendCtx.
+func (bc *litBackendCtx) FinishAndUnregisterEngines() error {
 	bc.unregisterMu.Lock()
 	defer bc.unregisterMu.Unlock()
 
 	if len(bc.engines) == 0 {
-		return
+		return nil
 	}
 	numIdx := int64(len(bc.engines))
+	uniqueIdx := make([]int64, 0, numIdx)
```

Comment:
Is it necessary to use a separate slice?

Reply:
Do you mean we call `collectRemoteDuplicateRows` inside the `	for _, ei := range bc.engines {` loop?

Reply:
Or use another `for _, ei := range bc.engines` after cleaning.

Reply:
https://github.com/pingcap/tidb/pull/54382/commits/ab2bccdf1ec0fa51a5aa3c1dcf95efd39698ee42

Reply:
https://github.com/pingcap/tidb/pull/54382/commits/7f13fdf98d875bacb3aa85f2e9ad002dcaabaac8

previous commit is wrong...